apps/worker/src/app/workflow/usecases/run-job/run-job.usecase.ts

```diff
@@ -114,19 +121,77 @@ export class RunJob {
     }
   }
 
+  private assignLogger(job) {
+    try {
+      this.logger?.assign({
+        transactionId: job.transactionId,
+        environmentId: job._environmentId,
+        organizationId: job._organizationId,
+        jobId: job._id,
+        jobType: job.type,
+      });
+    } catch (e) {
+      Logger.error(e, 'RunJob', LOG_CONTEXT);
+    }
+  }
+
+  /*
+   * If the following condition is met,
+   * - transactions were merged to the main delayed digest.
+   * - the main delayed digest was canceled.
+   * that mean that we need to assign a new active digest follower job to replace it.
+   * so from now on we will continue the follower transaction as main digest job.
+   */
+  private assignNewDigestExecutor(activeDigestFollower: JobEntity): JobEntity {
+    return activeDigestFollower;
+  }
+
+  private isCanceledMainDigest(type: StepTypeEnum | undefined, status: JobStatusEnum) {
+    return type === StepTypeEnum.DIGEST && status === JobStatusEnum.CANCELED;
+  }
+
   @Instrument()
-  private async delayedEventIsCanceled(job: JobEntity): Promise<boolean> {
+  private async delayedEventIsCanceled(
+    job: JobEntity
+  ): Promise<{ canceled: boolean; activeDigestFollower: JobEntity | null }> {
+    let activeDigestFollower: JobEntity | null = null;
+
     if (job.type !== StepTypeEnum.DIGEST && job.type !== StepTypeEnum.DELAY) {
-      return false;
+      return { canceled: false, activeDigestFollower };
     }
 
-    const count = await this.jobRepository.count({
+    const canceled = job.status === JobStatusEnum.CANCELED;
+
+    if (job.status === JobStatusEnum.CANCELED) {
+      activeDigestFollower = await this.activeDigestMainFollowerExist(job);
+    }
+
+    return { canceled, activeDigestFollower };
+  }
+
+  @Instrument()
+  private async activeDigestMainFollowerExist(job: JobEntity): Promise<JobEntity | null> {
+    if (job.type !== StepTypeEnum.DIGEST) {
+      return null;
+    }
+
+    const { digestKey, digestValue } = getJobDigest(job);
```

Comment:
after we moved the getNestedValue getJobDigest function into separated stateless functions now we can reuse getJobDigest here.